src/core/src/sys/new_system.rs

```diff
@@ -0,0 +1,257 @@
+    pub velocities: &'a mut [Vector3D],
+    pub molecule_ids: &'a mut [usize]
+}
+
+
+#[derive(Clone)]
+pub struct SystemGeometry {
+    particles: Particles,
+    cell: UnitCell,
+    molecules: Vec<Molecule>,
+}
+
+#[derive(Clone)]
+pub struct System {
+    geometry: SystemGeometry,
+    interactions: Interactions,
+    step: u64,
+    external_temperature: Option<f64>
+}
+
+impl Default for Particles {
+    fn default() -> Particles {
+        Particles {
+            names: Vec::new(),
+            kinds: Vec::new(),
+            masses: Vec::new(),
+            charges: Vec::new(),
+            positions: Vec::new(),
+            velocities: Vec::new(),
+            molecule_ids: Vec::new(),
+        }
+    }
+}
+
+impl Particles {
+    #[inline] pub fn len(&self) -> usize {
+        self.kinds.len()
+    }
+
+    pub fn as_slices(&self) -> ParticlesRef {
+        ParticlesRef {
+            names: &self.names,
+            kinds: &self.kinds,
+            masses: &self.masses,
+            charges: &self.charges,
+            positions: &self.positions,
+            velocities: &self.velocities,
+            molecule_ids: &self.molecule_ids
+        }
+    }
+
+    pub fn as_slices_mut(&mut self) -> ParticlesRefMut {
+        ParticlesRefMut {
+            names: &mut self.names,
+            kinds: &mut self.kinds,
+            masses: &mut self.masses,
+            charges: &mut self.charges,
+            positions: &mut self.positions,
+            velocities: &mut self.velocities,
+            molecule_ids: &mut self.molecule_ids
+        }
+    }
+
+
+    pub fn push(&mut self, particle: Particle, molecule_id: usize) {
+        self.names.push(particle.name);
+        self.kinds.push(particle.kind);
+        self.masses.push(particle.mass);
+        self.charges.push(particle.charge);
+        self.positions.push(particle.position);
+        self.velocities.push(particle.velocity);
+        self.molecule_ids.push(molecule_id);
+    }
+}
+
+impl<'a> ParticlesRef<'a> {
+    #[inline] pub fn len(&self) -> usize {
+        self.kinds.len()
+    }
+}
+
+impl<'a> ParticlesRefMut<'a> {
+    #[inline] pub fn len(&self) -> usize {
+        self.kinds.len()
+    }
+}
+
+impl Default for SystemGeometry {
+    fn default() -> SystemGeometry {
+        SystemGeometry {
+            particles: Particles::default(),
+            cell: UnitCell::new(),
+            molecules: Vec::new()
+        }
+    }
+}
+
+impl SystemGeometry {
+    pub fn particles(&self) -> ParticlesRef {
+        self.particles.as_slices()
+    }
+
+    pub fn particles_mut(&mut self) -> ParticlesRefMut {
+        self.particles.as_slices_mut()
+    }
+}
+
+
+impl Default for System {
+    fn default() -> System {
+        System {
+            geometry: SystemGeometry::default(),
+            interactions: Interactions::new(),
+            step: 0,
+            external_temperature: None
+        }
+    }
+}
+
+impl System {
+    pub fn from_cell(cell: UnitCell) -> System {
```

Comment:
This should really be called `with_cell`. I believe you just reused the name from the current code, right?